pyriemann_qiskit/classification.py

```diff
@@ -743,3 +750,327 @@ def predict(self, X):
+                self.matrices_per_class_[c][indexes], test_sample
+            )
+
+            if self.debug:
+                print("Final hull distance:", d)
+
+            distances.append(d)
+
+        return distances
+
+    def _process_sample_random_hull(self, test_sample):
+        """Uses random samples to build a hull, can be several hulls per class"""
+        distances = []
+
+        for c in self.classes_:
+            total_distance = 0
+
+            # using multiple hulls
+            for i in range(0, self.n_hulls_per_class):
+                if self.n_samples_per_hull == -1:  # use all data per class
+                    hull_data = self.matrices_per_class_[c]
+                else:  # use a subset of the data per class
+                    random_samples = random.sample(
+                        range(self.matrices_per_class_[c].shape[0]),
+                        k=self.n_samples_per_hull,
+                    )
+                    hull_data = self.matrices_per_class_[c][random_samples, :, :]
+
+                distance = qdistance_logeuclid_to_convex_hull(hull_data, test_sample)
+                total_distance = total_distance + distance
+
+            distances.append(total_distance)
+
+        return distances
+
+    def _predict_distances(self, X):
+        """Helper to predict the distance. Equivalent to transform."""
+        dist = []
+
+        if self.debug:
+            print("Total test samples:", X.shape[0])
+
+        if self.hull_type == "min-hull":
+            self._process_sample = self._process_sample_min_hull
+        elif self.hull_type == "random-hull":
+            self._process_sample = self._process_sample_random_hull
+        else:
+            raise Exception("Error: Unknown hull type.")
+
+        parallel = self.n_jobs > 1
+
+        if self.debug:
+            if parallel:
+                print("Running in parallel")
+            else:
+                print("Not running in parallel")
+
+        if parallel:
+            dist = Parallel(n_jobs=self.n_jobs)(
+                delayed(self._process_sample)(test_sample) for test_sample in X
+            )
+
+        else:
+            for test_sample in X:
+                dist_sample = self._process_sample(test_sample)
+                dist.append(dist_sample)
+
+        return dist
+
+    def predict(self, X):
+        """Get the predictions.
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        Returns
+        -------
+        pred : ndarray of int, shape (n_matrices,)
+            Predictions for each matrix according to the closest convex hull.
+        """
+        if self.debug:
+            print("Start NCH Predict")
+        dist = self._predict_distances(X)
+
+        predictions = [
+            self.classes_[min(range(len(values)), key=values.__getitem__)]
+            for values in dist
+        ]
```

Comment:
What is the difference with `predictions = self.classes_[dist.argmin(axis=1)]` ?